src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java

```diff
@@ -114,6 +118,9 @@ public static void finalizeRegion(final AssemblyRegion region,
                 .filter(read ->  !read.isEmpty() && read.getCigar().getReadLength() > 0)
                 .map(read -> ReadClipper.hardClipToRegion(read, region.getPaddedSpan().getStart(), region.getPaddedSpan().getEnd() ))
                 .filter(read -> read.getStart() <= read.getEnd() && read.getLength() > 0 && read.overlaps(region.getPaddedSpan()))
+                // The transient attribute is preserved across copy operations and all of the previous alterations make copies, this simple ensures
+                // that any reads that have not been copied along the way are copied here for safety.
+                .map(read -> (read.getTransientAttribute("Original") != read? read : read.copy()))
```

Comment:
Instead of relying on transient attributes (which, in my opinion, is brittle and a recipe for future problems), compare the memory address of the read pre/post clipping, and make a copy only if the address hasn't changed, as in the original patch in https://github.com/broadinstitute/gatk/pull/4926. Rewrite this method to use a loop instead of streaming, if necessary.

Reply:
I have rewritten it as a loop at the expense of angering the Aesthetic gods. Can you re-review?